rpm/0022-brp-python-bytecompile-Ensure-reproducibility-of-inv.patch

```diff
@@ -0,0 +1,31 @@
+From 3e62497b0fa9bc30378a9f280cc7a7ef9d9f5aac Mon Sep 17 00:00:00 2001
+From: Martin Kampas <martin.kampas@jolla.com>
+Date: Wed, 7 Sep 2022 18:30:02 +0200
+Subject: [PATCH] brp-python-bytecompile: Ensure reproducibility of
+ invalidation timestamp
+
+---
+ scripts/brp-python-bytecompile | 7 +++++++
+ 1 file changed, 7 insertions(+)
+
+diff --git a/scripts/brp-python-bytecompile b/scripts/brp-python-bytecompile
+index 97691935d..345411f48 100644
+--- a/scripts/brp-python-bytecompile
++++ b/scripts/brp-python-bytecompile
+@@ -61,6 +61,13 @@ EOF
+ # This helps to make byte-compilation more reproducible
+ export PYTHONHASHSEED=0
+ 
++# .pyc/.pyo files embed the time of modification of the source file. Ensure
++# build reproducibility by adjusting the modification time a reproducible way.
++if [ -n "$SOURCE_DATE_EPOCH" ]; then
++	echo "Updating modification time of .py files according to \$SOURCE_DATE_EPOCH"
++	find "$RPM_BUILD_ROOT" -name '*.py' -print0 |xargs --verbose -0 touch --date=@"$SOURCE_DATE_EPOCH"
```

Comment:
-exec instead of xargs would work here too.

Reply:
`-exec` would spawn `touch` for each file, `xargs` spawns once for `ARG_MAX`.